### DISCUSSION TITLE
InternalFederateException and NullPointerException

### DISCUSSION THREAD

**Discussion by babangidazachariah** [eclipse/mosaic#386] [General]
What could cause the following error and how to address it:

        - InternalFederateException: Error code: 0x01000013: There was an error during an advance time.
        - Root Cause: NullPointerException: The application has not been initialized by the ambassador: the OperatingSystem is null.

**Comment by schwepmo** [comment]
This probably is an issue with your application but without any context it is hard to pinpoint the issue. When writing an application for a vehicle make sure that your application extends the `AbstractApplication<T>` class using  a `VehicleOperatingSystem`.

For additional reference see here: https://eclipse.dev/mosaic/docs/develop_applications/

**Reply by babangidazachariah** [reply]
Thank you sir. I was able to figure it out. I was trying to instantiate objects of my classes in the onStartup() method, the error was or appeared to have been fixed after I moved the instantiations into another function and creating a schedule event to execute the function after the simulation has started. This I don't know why though.

Thank you

**Reply by schwepmo** [reply]
Alright, I'm glad you found a solution for your problem. If your classes require road positions, this could cause the issue. These are only available after `onVehicleUpdated` has been called. A common pattern we find in our applications is to call some initialize methods within the `onVehicleUpdated`-method with the condition `if (previousVehicleData == null) { ... }` as this will only happen once in the beginning of the simulation

**Reply by babangidazachariah** [reply]
I see. Thank you @schwepmo I really appreciate
